pyts/metrics/dtw.py

```diff
@@ -526,40 +543,65 @@ def itakura_parallelogram(n_timestamps, max_slope=2.):
     --------
     >>> from pyts.metrics import itakura_parallelogram
     >>> print(itakura_parallelogram(5))
-    [[0 1 1 2 4]
-     [1 3 4 4 5]]
+    [[0 0 1 2 4]
+     [1 3 4 5 5]]
 
     """
-    if not isinstance(n_timestamps, (int, np.integer)):
-        raise TypeError("'n_timestamps' must be an intger.")
+    if not isinstance(n_timestamps_1, (int, np.integer)):
+        raise TypeError("'n_timestamps_1' must be an integer.")
     else:
-        if not n_timestamps >= 2:
-            raise ValueError("'n_timestamps' must be an integer greater than "
-                             "or equal to 2.")
+        if not n_timestamps_1 >= 2:
+            raise ValueError("'n_timestamps_1' must be an integer greater than"
+                             " or equal to 2.")
+    if n_timestamps_2 is None:
+        n_timestamps_2 = n_timestamps_1
+
     if not isinstance(max_slope, (int, np.integer, float, np.floating)):
         raise TypeError("'max_slope' must be an integer or a float.")
     else:
         if not max_slope >= 1:
             raise ValueError("'max_slope' must be a number greater "
                              "than or equal to 1.")
 
+    # rescale max_slop to the relative lengths slope
+
     min_slope = 1 / max_slope
+    scale = (n_timestamps_2 - 1) / (n_timestamps_1 - 1)
+    min_slope *= min(scale, 1 / scale)
+    max_slope *= max(scale, 1 / scale)
+
+    # Now we create the piecewise linear functions defining the parallelogram
+
+    # lower_bound[0] = min_slope * x
+    # lower_bound[1] = max_slope * (x - n_timestamps_1) + n_timestamps_2
+
+    centered_scale = np.arange(n_timestamps_1) - n_timestamps_1 + 1
+    lower_bound = np.empty((2, n_timestamps_1))
+    lower_bound[0] = min_slope * np.arange(n_timestamps_1)
+    lower_bound[1] = max_slope * centered_scale + n_timestamps_2 - 1
+
+    # take the max of the lower linear funcs
+    lower_bound = np.max(np.floor(lower_bound), axis=0)
 
-    lower_bound = np.empty((2, n_timestamps))
-    lower_bound[0] = min_slope * np.arange(n_timestamps)
-    lower_bound[1] = ((1 - max_slope) * (n_timestamps - 1)
-                      + max_slope * np.arange(n_timestamps))
-    lower_bound = np.round(lower_bound, 2)
-    lower_bound = np.ceil(np.max(lower_bound, axis=0))
+    # upper_bound[0] = max_slope * x
```

Comment:
I added them to make the code clearer for future collaborators (who could be us :) ) because it took me a while to figure out what `lower_bound` and `upper_bound` refer to

Reply:
Indeed, it's a good idea (there aren't many comments in my code...). Maybe adding a short sentence could be helpful, something like:
*'lower_bounds' and 'upper_bounds' are the equations defining the sides of the parallelogram*